infra/base-images/base-clang/checkout_build_install_llvm.sh

```diff
@@ -98,6 +98,7 @@ function cmake_llvm {
       -DLIBCXX_ENABLE_SHARED=OFF \
       -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=ON \
       -DLIBCXXABI_ENABLE_SHARED=OFF \
+      -DLIBCXX_ENABLE_DEBUG_MODE=ON \
```

Comment:
Note that if Clang gets rolled to 18.x (#11714) , then we'll want to switch these for `LIBCXX_HARDENING_MODE`: https://libcxx.llvm.org/Hardening.html.